### ISSUE TITLE
Multiple redirect infinite loop when username and old_old_username are the same.

### ISSUE THREAD

**Issue by benhalpern** [forem/forem#12444]
**Describe the bug**

We persist username redirects back two levels, but it's possible for organizations and users can have the same `username` and `old_old_username` if they change to one and then change back. When those two are the same, it can cause infinite redirects in certain specific situations.

**To Reproduce**

Visit this link: https://dev.to/sarthology/do-we-really-need-a-css-framework-4ma6

It causes a redirect.

**Expected behavior**

This should ultimately redirect to https://dev.to/xenoxdev/do-we-really-need-a-css-framework-4ma6 but breaks due to this org having `xenoxdev` as both current slug and `old_old_slug`.

Expected behavior on saving a new username is that it would erase a conflicting `old_old_...` version, as it is implied that it's no longer relevant.

**Additional context**

This is attribute called `username` for users and `slug` for orgs.

It seems like this could be fixed by checking if the new `username`/`slug` is the same as the `old_old_username`/`old_old_slug` and clearing the `old_old_...` in that case before save.

**Urgency**

This is causing current user problems and is also penalized by SEO.

**Comment by citizen428** [comment]
tl;dr:

1. This only seems to happen to articles that belong to organizations but are accessed via the author's slug.
2. I'm not yet sure if the best possible fix is the one regarding `old_username` and `old_old_username` described in the original issue or if we want to update the logic in `StoriesController#show` instead.

More context below.
---

I had a bit of a hard time replicating this:

I had a local user that I set up as described:

```ruby
[user.username, user.old_username, user.old_old_username]
#=> ["admin_mcadmin", "bossmang", "admin_mcadmin"]
```

Yet everything seemed to be working as expected.

When looking at the live data in Blazer:

* The user `sarthology` has a username but no `old_username` or `old_old_username`
* The organization `xenoxdev` has and `old_slug` of `teamxenox` and an `old_old_slug` of `xenoxdev`

It's wasn't immediately obvious to me what caused the initial redirect from the user to the organization until I realized that the article belongs to the organization and we redirect because of the `elsif` branch in `StoriesController#handle_possible_redirect`:

```ruby
    elsif (@organization = @article.organization)
      redirect_permanently_to(URI.parse("/#{@organization.slug}/#{params[:slug]}").path)
      return
    end
```

I then had a look at the `show` action which made me finally realize why I couldn't replicate this:

```ruby
    if (@article = Article.find_by(path: "/#{params[:username].downcase}/#{params[:slug]}")&.decorate)
      handle_article_show
    elsif (@article = Article.find_by(slug: params[:slug])&.decorate)
      handle_possible_redirect
    else
```

Articles for organizations don't have a slug of the form `/:username/:slug` but /:organizartion/:slug` instead. However, my test article didn't belong to an organization, so everything worked. In the case of the `sarthology` / `xenoxdev` problem we can't find the article, so go the `handle_possible_redirect` route instead.

**Comment by citizen428** [comment]
Confirmed that #12652 really fixed the issue, closing this issue now.
